### PR TITLE
Intercept for missing business info and  account info changes

### PR DESCRIPTION
**Pull request by saravanpa-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/7783

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/components/auth/account-settings/account-info/AccountInfo.vue [hunk 1]
```diff
@@ -9,15 +9,6 @@
...
 
-        <div v-can:VIEW_ACCOUNT.hide>
-          <div class="nv-list-item mb-10">
-            <div class="name" id="accountName">Account Name</div>
-            <div class="value" aria-labelledby="accountName">
-              <div class="value__title">{{ currentOrganization.name }}</div>
-            </div>
-          </div>
-        </div>
-
```
Comment (saravanpa-aot) [inline auth-web/src/components/auth/account-settings/account-info/AccountInfo.vue:20]: this code block was for showing name to USER/COORDINATOR who couldn't change the name...now name is disabled for everyone..so no need of it I think

File: auth-web/src/components/auth/account-settings/account-info/AccountInfo.vue [hunk 2]
```diff
@@ -90,23 +81,37 @@
           </div>
         </div>
 
-        <div class="nv-list-item" v-can:CHANGE_ORG_NAME.disable>
```
Comment (saravanpa-aot) [inline auth-web/src/components/auth/account-settings/account-info/AccountInfo.vue:93]: name is always disabled

**Review by karthik-aot — APPROVED** [review]

File: auth-web/src/components/auth/common/AccountBusinessTypePicker.vue [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+import { Code } from '@/models/Code'
+import OrgNameAutoComplete from '@/views/auth/OrgNameAutoComplete.vue'
+import { namespace } from 'vuex-class'
+
+const OrgModule = namespace('org')
+const CodesModule = namespace('codes')
+
+@Component({
+  components: {
+    OrgNameAutoComplete
```
Comment (shabeeb-aot) [inline auth-web/src/components/auth/common/AccountBusinessTypePicker.vue:64]: looks like this is not using

File: auth-web/src/components/auth/common/AccountBusinessTypePicker.vue [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    businessType: HTMLFormElement,
+    businessSize: HTMLFormElement
+  }
+
+  // input fields
+  private isBusinessAccount = false
+  private name = ''
+  private businessType = ''
+  private businessSize = ''
+  private branchName = ''
```
Comment (shabeeb-aot) [inline auth-web/src/components/auth/common/AccountBusinessTypePicker.vue:90]: this component contain only businessType and businessSize. are we using name and branchName

File: auth-web/src/components/auth/common/AccountBusinessTypePicker.vue [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+  @Emit('update:org-business-type')
+  private emitUpdatedOrgBusinessType () {
+    const orgBusinessType: OrgBusinessType = { businessType: this.businessType, businessSize: this.businessSize }
+    return orgBusinessType
+  }
+
+  /** Emits the validity of the component. */
+  @Emit('valid')
+  private emitValid () {
+    return !this.$refs.businessType?.hasError && !this.$refs.businessSize?.hasError
```
Comment (shabeeb-aot) [inline auth-web/src/components/auth/common/AccountBusinessTypePicker.vue:106]: can we use one ref and and use validate?

File: auth-web/src/components/auth/common/AccountBusinessTypePicker.vue [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+      // load business type and size codes
+      this.isLoading = true
+      await this.getBusinessSizeCodes()
+      await this.getBusinessTypeCodes()
+      if (this.currentOrganization.name) {
+        // incase if the new account is a premium account, default business type to business account
+        this.isBusinessAccount = this.currentOrganization.isBusinessAccount
+        this.businessType = this.currentOrganization.businessType
+        this.businessSize = this.currentOrganization.businessSize
+        this.branchName = this.currentOrganization.branchName
```
Comment (shabeeb-aot) [inline auth-web/src/components/auth/common/AccountBusinessTypePicker.vue:124]: here also

File: auth-web/src/components/auth/common/AccountBusinessTypePicker.vue [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+      } else {
+        this.isBusinessAccount = this.currentOrganization.orgType !== Account.BASIC
```
Comment (shabeeb-aot) [inline auth-web/src/components/auth/common/AccountBusinessTypePicker.vue:126]: we are check isBusinessAccount from parent itself?

File: auth-web/src/components/auth/common/AccountBusinessTypePicker.vue [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+@import '$assets/scss/theme.scss';
+.view-container {
+    padding: 0 !important;
+}
+.business-radio{
```
Comment (shabeeb-aot) [inline auth-web/src/components/auth/common/AccountBusinessTypePicker.vue:151]: can remove this

File: auth-web/src/views/auth/create-account/UpdateAccountView.vue [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+      </v-col>
+    </v-row>
+    <v-row justify="center">
+      <v-col cols="12" sm="9" md="8" class="text-center">
+        <v-card flat class="my-4 justify-space-between pa-8">
+          <v-card-title>
+            <h2>Account Information</h2>
+          </v-card-title>
+          <v-card-text>
+            <p style="text-align: justify">
```
Comment (shabeeb-aot) [inline auth-web/src/views/auth/create-account/UpdateAccountView.vue:18]: use class than direct style

File: auth-web/src/views/auth/create-account/UpdateAccountView.vue [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+import { CreateRequestBody, OrgBusinessType, Organization } from '@/models/Organization'
+import AccountBusinessType from '@/components/auth/common/AccountBusinessType.vue'
+import AccountBusinessTypePicker from '@/components/auth/common/AccountBusinessTypePicker.vue'
+import { Pages } from '@/util/constants'
+import { namespace } from 'vuex-class'
+
+const OrgModule = namespace('org')
+
+@Component({
+  components: { AccountBusinessTypePicker, AccountBusinessType }
```
Comment (shabeeb-aot) [inline auth-web/src/views/auth/create-account/UpdateAccountView.vue:102]: are we using AccountBusinessType in this component

File: auth-web/src/views/auth/create-account/UpdateAccountView.vue [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+})
+export default class UpdateAccountView extends Vue {
+  @OrgModule.State('currentOrganization') private currentOrganization!: Organization
+  @OrgModule.Action('syncOrganization') private syncOrganization!: (orgId: number) => Promise<Organization>
+  @OrgModule.Action('isOrgNameAvailable') private readonly isOrgNameAvailable!: (orgName: string) => Promise<boolean>
```
Comment (shabeeb-aot) [inline auth-web/src/views/auth/create-account/UpdateAccountView.vue:107]: looks like not using

File: auth-web/src/views/auth/create-account/UpdateAccountView.vue [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+import { namespace } from 'vuex-class'
+
+const OrgModule = namespace('org')
+
+@Component({
+  components: { AccountBusinessTypePicker, AccountBusinessType }
+})
+export default class UpdateAccountView extends Vue {
+  @OrgModule.State('currentOrganization') private currentOrganization!: Organization
+  @OrgModule.Action('syncOrganization') private syncOrganization!: (orgId: number) => Promise<Organization>
```
Comment (shabeeb-aot) [inline auth-web/src/views/auth/create-account/UpdateAccountView.vue:106]: looks like not using

**Review by shabeeb-aot — APPROVED** [review]

**Pull request merged**